pkg/ccl/backupccl/backup_processor.go

```diff
@@ -608,24 +620,27 @@ func (s *sstSink) push(ctx context.Context, resp returnedSST) error {
 	s.queue = append(s.queue, resp)
 	s.queueSize += len(resp.sst)
 
-	if len(s.queue) >= maxSinkQueueFiles || s.queueSize >= int(smallFileBuffer.Get(s.conf.settings)) {
+	if s.queueSize >= int(smallFileBuffer.Get(s.conf.settings)) {
 		sort.Slice(s.queue, func(i, j int) bool { return s.queue[i].f.Span.Key.Compare(s.queue[j].f.Span.Key) < 0 })
 
 		// Drain the first half.
 		drain := len(s.queue) / 2
 		if drain < 1 {
 			drain = 1
 		}
+		var drainedByteSize int
 		for i := range s.queue[:drain] {
 			if err := s.write(ctx, s.queue[i]); err != nil {
 				return err
 			}
 			s.queueSize -= len(s.queue[i].sst)
+			drainedByteSize += len(s.queue[i].sst)
 		}
 
 		// Shift down the remainder of the queue and slice off the tail.
 		copy(s.queue, s.queue[drain:])
 		s.queue = s.queue[:len(s.queue)-drain]
+		s.backupMem.release(int64(drainedByteSize))
```

Comment:
Actually, I'm a little confused why we `release` here, as we drain, given that we don't `request` above when we push?

Also, I think we could just directly call release above, instead of accumulating `drainedByteSize` here, right?

Reply:
You're right, this is a mistake. We're growing it once when we're initializing the sstSink, and then only closing it when the backup processor is closing. This then begs the question of whether we even need the `memoryAccumulator`. 

Even if we end up tying an sstSink to a worker in the future, we can just wrap the backup processors memMonitor/boundAccount in a mutex.

Reply:
I decided to keep the memAccumulator, if we decide to mem monitor other parts of the backup processor itll be nice to have this shared pool to take from. This was also heavily inspired by Pauls' restore mem monitor PR so when we do get to that we might be able to reuse the util.

I moved the request call to a `makeSSTSink` method and added a `release` call to the close sink method.